desc/objectives/_generic.py

```diff
@@ -248,4 +262,733 @@ def compute(self, R_lmn, Z_lmn, L_lmn, i_l, Psi, **kwargs):
             self._iota,
         )
         I = 2 * jnp.pi / mu_0 * data["I"]
-        return self._shift_scale(jnp.atleast_1d(I))
+        return self._shift_scale(I)
+
+
+class MercierStability(_Objective):
+    """
+    The Mercier criterion is a fast proxy for MHD stability.
+    This makes it a useful figure of merit for stellarator operation.
+    Systems with DMerc > 0 are favorable for stability.
+
+    See equation 4.16 in
+    Landreman, M., & Jorge, R. (2020). Magnetic well and Mercier stability of
+    stellarators near the magnetic axis. Journal of Plasma Physics, 86(5), 905860510.
+    doi:10.1017/S002237782000121X.
+    """
+
+    _scalar = True
+    _linear = False
+
+    def __init__(self, eq=None, target=0, weight=1, grid=None, name="Mercier DMerc"):
+        """Initialize a Mercier Stability Objective.
+
+        Parameters
+        ----------
+        eq : Equilibrium, optional
+            Equilibrium that will be optimized to satisfy the Objective.
+        target : float, ndarray, optional
+            Target value(s) of the objective.
+            len(target) must be equal to Objective.dim_f
+        weight : float, ndarray, optional
+            Weighting to apply to the Objective, relative to other Objectives.
+            len(weight) must be equal to Objective.dim_f
+        grid : LinearGrid, ConcentricGrid, QuadratureGrid, ndarray, optional
+            Collocation grid containing the nodes to evaluate at.
+            Due to the symmetry / NFP bugs, the grid should temporarily be limited to pass this assertion:
+                assert (grid.num_rho == 1) or (grid.sym is False and grid.NFP == 1)
+        name : str
+            Name of the objective function.
+        """
+        self.grid = grid
+        super().__init__(eq=eq, target=target, weight=weight, name=name)
+        self._callback_fmt = "Mercier DMerc: {:10.3e}"
+
+    def build(self, eq, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        eq : Equilibrium, optional
+            Equilibrium that will be optimized to satisfy the Objective.
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+        """
+        if self.grid is None:
```

Comment:
is this a good default grid? only at rho=1?

Reply:
I think that's the default for all the objectives in generic. I should delete this comment from all these docstrings now that the symmetry and NFP bugs are fixed. `Due to the symmetry / NFP bugs, the grid should temporarily be limited...`